logger/logger.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package logger has functions to obtain a logger, and helpers for setting up where the logger writes
+package logger
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sonatype-nexus-community/nancy/types"
+)
+
+// DefaultLogFile can be overriden to use a different file name for upstream consumers
+var DefaultLogFile = "nancy.combined.log"
+
+// Logger can be obtained from outside the package
+var Logger = logrus.New()
+
+func init() {
+	file, err := os.OpenFile(getLogFileLocation(), os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0666)
+	if err != nil {
+		fmt.Print(err)
+	}
+
+	Logger.Out = file
+	Logger.Level = logrus.InfoLevel
+	Logger.Formatter = &logrus.JSONFormatter{}
```

Comment:
Is it worthwhile to create a test around this to make sure its actually configured to output a file and put it in the `.ossindex` dir??

Reply:
PROBABLY

Reply:
Test here: 78e6990